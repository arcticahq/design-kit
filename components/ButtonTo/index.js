import React from "react"
import PropTypes from "prop-types"
import { Link } from "gatsby"
import Ink from "react-ink"
import classnames from "classnames"

const ButtonTo = ({ href, to, children, className, isInline, primary }) => {
  const buttonClass = classnames({
    "relative text-black text-xs bg-white py-3 px-4 rounded-full font-semibold uppercase rise hover:text-black": true,
    "inline-block": !isInline,
    inline: isInline,
    "bg-indigo-600 text-white hover:text-white": primary,
    [className]: className && true
  })

  if (href) {
    return (
      <a
        href={href}
        target="_blank"
        rel="external noreferrer noopener"
        className={buttonClass}
      >
        <Ink />
        {children}
      </a>
    )
  }
  return (
    <Link to={to} className={buttonClass}>
      <Ink />
      {children}
    </Link>
  )
}

ButtonTo.propTypes = {
  href: PropTypes.string,
  to: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.string.isRequired,
  isInline: PropTypes.bool,
  primary: PropTypes.bool
}

ButtonTo.defaultProps = {
  href: null,
  to: null,
  isInline: false,
  primary: false,
  className: null
}

export default ButtonTo
