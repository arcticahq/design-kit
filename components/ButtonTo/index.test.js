import React from "react"
import { shallow } from "enzyme"
import ButtonTo from "."

describe(`ButtonTo`, () => {
  let props
  let mountedComponent
  const wrapper = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<ButtonTo {...props}>Contact</ButtonTo>)
    }
    return mountedComponent
  }

  beforeEach(() => {
    props = {
      to: `/contact/`
    }
    mountedComponent = undefined
  })

  it(`renders a ButtonTo correctly`, () => {
    expect(wrapper()).toMatchSnapshot()
  })
})
