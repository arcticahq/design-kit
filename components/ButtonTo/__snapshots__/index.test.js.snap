// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTo renders a ButtonTo correctly 1`] = `
<mockConstructor
  className="relative text-black text-xs bg-white py-3 px-4 rounded-full font-semibold uppercase rise hover:text-black inline-block"
  to="/contact/"
>
  <e
    background={true}
    className="ink"
    duration={1000}
    opacity={0.25}
    radius={150}
    recenter={true}
  />
  Contact
</mockConstructor>
`;
